### PR TITLE
Add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by tsmartt**
This is meant to be a set of guidelines for anyone contributing to follow, and this PR is meant to spark constructive criticism.
Please feel free to propose additions where any might be lacking, and modifications with good justification.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+- Follow the [code review guildelines][codereview]
+- Write tests to cover critical functionality
+- Write a [good commit message][commit]
+- Leave code better than you found it, whenever possible:
+  - eliminate [callbacks][callbacks], use a service object instead
+  - use [dependency injection][di] to encourage modularity, [an example][di-ex]
+  - follow Sandi Metz' [rules][sm]
+  - prefer [composition over inheritance][composition]
+
+[composition]: https://betterprogramming.pub/prefer-composition-over-inheritance-1602d5149ea1
```
Comment (yachtcaptain23) [inline CONTRIBUTING.md:28]: IMO the article is dogmatic, other's like such as https://thoughtbot.com/blog/reusable-oo-composition-vs-inheritance suggest that the developer should understand the trade-off between composition vs inheritance.

**Comment by tsmartt** [comment]
Another thought I had was mimicking the behavior of [mutations](https://github.com/cypriss/mutations) and [interactor](https://github.com/collectiveidea/interactor#an-example-interactor), but which a much simpler setup, something like:

```ruby
# frozen_string_literal: true
module My
    class Service < BaseService
        class COMMAND
            include ActiveModel::Validations
            validates_presence_of :param1
            validates_presence_of :param2
        end
      def self.build
        new(tp_need: TPNeed)
      end
      def initialize(tp_need:)
        @tp_need = tp_need
      end
      def execute(param1:, param2)
        result = @tp_need.do_something(param1, param2)
        if not result
            return fail("Some error message")
        end
        success("Did thing")
      end
    end
  end
end
```

The base class would be something like 40 lines of ruby and would give us what we need with no extra frameworks. It instantiates the COMMAND class with the parameters and validates, if validation passes, it calls execute.
By passing the params both to self.build and execute, we are free to use ruby's type system features to help stabilize the code. To invoke it you just `My::Service.build.call(param1: '1', param2: '2')`

For most things you could probably even leave out the COMMAND -- since the type system could take care of basic things like non-nil, etc.. could add it in for more complex scenarios like email validation.

File: CONTRIBUTING.md (hunk 1 above)
Comment (tsmartt) [reply CONTRIBUTING.md:28]: I agree, that's why I put prefer.

**Comment by tsmartt** [comment]
@yachtcaptain23 going to merge this tomorrow unless any further thoughts.

**Review by yachtcaptain23 — APPROVED** [review]

**Pull request merged**